packages/riipen-ui/src/components/Popover.jsx

```diff
@@ -0,0 +1,300 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import clsx from "clsx";
+
+import ThemeContext from "../styles/ThemeContext";
+
+import {
+  getContainer,
+  getDocument,
+  getOffsetLeft,
+  getOffsetTop,
+  debounce
+} from "../utils";
+
+class Popover extends React.Component {
+  static propTypes = {
+    /**
+     * The content of the component
+     */
+    children: PropTypes.object,
+
+    /**
+     * Array or string of additional CSS classes to use.
+     *
+     * @type {string | Array}
+     */
+    classes: PropTypes.oneOfType([PropTypes.array, PropTypes.string]),
+
+    /**
+     * Function call to handle clickaway/ close events,
+     * if not provided the anchor element must be removed to clear the popover or use isOpen
+     */
+    onClose: PropTypes.func,
+
+    /**
+     * Either a reference to an anchor element or a function to get the reference
+     */
+    anchorEl: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
+
+    /**
+     * Whether the popover should be displayed
+     */
+    isOpen: PropTypes.bool,
+
+    /**
+     * The marigins of the page the popover should respect
+     */
+    marginThreshold: PropTypes.number,
+
+    /**
+     * The location to attach the content too on the anchor element
+     */
+    anchorPosition: PropTypes.object,
+
+    /**
+     * The location to attach the anchor to on the content element
+     */
+    contentPosition: PropTypes.object,
+
+    /**
+     * Whether to lock the scrollbar when the popover is open
+     */
+    lockScroll: PropTypes.bool,
```

Comment:
Added this so that its customizable whether or not to lock scrolling when the popover is open